### PR TITLE
Add more monai functionality

### PR DESCRIPTION
**Pull request by benjijamorris**

### CODE REVIEW

File: aicsmlsegment/DataLoader3D/Universal_Loader.py [hunk 1]
```diff
@@ -74,18 +124,16 @@ def __init__(
         input:
             filenames: path to images
             num_patch: number of random crops to be produced
-            size_in: size of input to model
             size_out: size of output from model
             n_channel: number of iput channels expected by model
             transforms: list of strings specifying transforms
+            patchize: whether to divide image into patches
```
Comment (jxchen01) [inline aicsmlsegment/DataLoader3D/Universal_Loader.py]: i love the name "patchize"

File: aicsmlsegment/DataLoader3D/Universal_Loader.py [hunk 2]
```diff
@@ -240,167 +271,232 @@ def __len__(self):
...
+    y_patch_sz = y_max // pr[1]
+    z_patch_sz = z_max // pr[0]
+
+    assert (
+        x_patch_sz > patch_size[2]
+        and y_patch_sz > patch_size[1]
+        and z_patch_sz > patch_size[0]
+    ), "Large image resize patches must be larger than model patch size"
+
+    # assert (
```
Comment (jxchen01) [inline aicsmlsegment/DataLoader3D/Universal_Loader.py]: do we need to keep these commented out code?

File: aicsmlsegment/bin/predict.py [hunk 3]
```diff
@@ -1,246 +1,43 @@
...
-            imsave(
-                config["OutputDir"]
-                + os.sep
-                + pathlib.PurePosixPath(fn).stem
-                + "_struct_segmentation.tiff",
-                out,
-            )
-
-            print(f"Image {fn} has been segmented")
+    model = Monai_BasicUNet.load_from_checkpoint(
```
Comment (jxchen01) [inline aicsmlsegment/bin/predict.py]: does this only support BasicUNet for now?

File: aicsmlsegment/bin/train.py [hunk 4]
```diff
@@ -19,24 +16,18 @@ def main():
...
     # create logger
     logger = get_logger("ModelTrainer")
-    config = load_config(args.config)
+    config, model_config = load_config(args.config, train=True)
     logger.info(config)
 
     # load a specified saved model
     if config["resume"] is not None:
         print(f"Loading checkpoint '{config['resume']}'...")
         model = Monai_BasicUNet.load_from_checkpoint(
```
Comment (jxchen01) [inline aicsmlsegment/bin/train.py]: i am not sure if I understand correctly. does this only support basicunet for now?

File: aicsmlsegment/bin/train.py [hunk 5]
```diff
@@ -19,24 +16,18 @@ def main():
...
     else:
         print("Training new model from scratch")
-        model = Monai_BasicUNet(config, train=True)
-
-    # Log the number of learnable parameters
-
-    logger.info(
-        f"Number of learnable params {get_number_of_learnable_parameters(model)}"
-    )
+        model = Monai_BasicUNet(config, model_config, train=True)
```
Comment (jxchen01) [inline aicsmlsegment/bin/train.py]: maybe we could try to use similar idea as in aics-segmentation, where modules are loaded based on input string?
https://github.com/AllenCell/aics-segmentation/blob/main/aicssegmentation/bin/batch_processing.py#L198

for example:
```
if config['model'] =="unet_xy_zoom':
    # use unet_xy_zoom,
elif config['model'] =="unet_xy":
    # use unet_xy just like before
else:
    # the other two are just for back compatibility, for new models users can set "model": "monai.basicunet" or any model that monai supports. then use importlib to load the specific monai module. 
```

would that work? I am not quite sure. thought?

File: aicsmlsegment/bin/predict.py (hunk 3 above)
Comment (benjijamorris) [reply aicsmlsegment/bin/predict.py]: Yes, should I add the 3 original models?

File: aicsmlsegment/bin/train.py (hunk 5 above)
Comment (benjijamorris) [reply aicsmlsegment/bin/train.py]: Yes, I think that would be smart

File: aicsmlsegment/bin/predict.py (hunk 3 above)
Comment (jxchen01) [reply aicsmlsegment/bin/predict.py]: yes, I think we should do similar thing like what we discuss below on `train.py
`

**Review by vianamp — APPROVED** [review]

**Review by sandraoluoch — APPROVED** [review]

**Pull request merged**